platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/DataFileMetadata.java

```diff
@@ -101,78 +101,92 @@ public DataFileMetadata(
      * @throws IOException If there was a problem reading metadata footer from the file
      */
     public DataFileMetadata(Path file) throws IOException {
-        try (final SeekableByteChannel channel = Files.newByteChannel(file, StandardOpenOption.READ)) {
-            // read footer from end of file
-            final ByteBuffer buf = ByteBuffer.allocate(FOOTER_SIZE);
-            channel.position(channel.size() - FOOTER_SIZE);
-            MerkleDbFileUtils.completelyRead(channel, buf);
-            buf.rewind();
-            // parse content
-            this.fileFormatVersion = buf.getInt();
-            this.dataItemValueSize = buf.getInt();
-            this.dataItemCount = buf.getLong();
-            this.index = buf.getInt();
-            this.creationDate = Instant.ofEpochSecond(buf.getLong(), buf.getInt());
-            buf.get(); // backwards compatibility: used to be a byte for isMergeFile
-            this.serializationVersion = buf.getLong();
+        MappedByteBuffer readingMmap = null;
+        try (final FileChannel channel = FileChannel.open(file, StandardOpenOption.READ)) {
+            readingMmap = channel.map(MapMode.READ_ONLY, 0, Math.min(1024, channel.size()));
```

Comment:
Extract 1024 to a constant, please

Reply:
This code is rewritten without hard-coded values in the recent versions of the PR.